src/contracts/AbstractARM.sol

```diff
@@ -0,0 +1,652 @@
+        IERC20(liquidityAsset).transfer(msg.sender, assets);
+    }
+
+    /// @notice Check if a withdrawal request can be claimed
+    /// @param requestId The index of the withdrawal request
+    function isClaimable(uint256 requestId) public view returns (bool) {
+        return
+            withdrawalRequests[requestId].queued <= withdrawsClaimed + IERC20(liquidityAsset).balanceOf(address(this));
+    }
+
+    /// @dev Calculate how much of the liquidity asset (WETH) in the ARM is not reserved for the withdrawal queue.
+    // That is, the amount of liquidity assets (WETH) that is available to be swapped.
+    // If there are no outstanding withdrawals, then return the maximum uint256 value.
+    // The ARM can swap out liquidity assets (WETH) that has been accrued from the performance fee for the fee collector.
+    // There is no liquidity guarantee for the fee collector. If there is not enough liquidity assets (WETH) in
+    // the ARM to collect the accrued fees, then the fee collector will have to wait until there is enough liquidity assets.
+    function _liquidityAvailable() internal view returns (uint256) {
+        // The amount of WETH that is still to be claimed in the withdrawal queue
+        uint256 outstandingWithdrawals = withdrawsQueued - withdrawsClaimed;
+
+        // Save gas on an external balanceOf call if there are no outstanding withdrawals
+        if (outstandingWithdrawals == 0) return type(uint256).max;
+
+        // The amount of the liquidity asset is in the ARM
+        uint256 liquidityBalance = IERC20(liquidityAsset).balanceOf(address(this));
+
+        // If there is not enough liquidity assets in the ARM to cover the outstanding withdrawals
+        if (liquidityBalance <= outstandingWithdrawals) {
+            return 0;
+        }
+
+        return liquidityBalance - outstandingWithdrawals;
+    }
+
+    /// @notice The total amount of assets in the ARM and external withdrawal queue,
+    /// less the liquidity assets reserved for the ARM's withdrawal queue and accrued fees.
+    function totalAssets() public view virtual returns (uint256) {
+        (uint256 fees, uint256 newAvailableAssets) = _feesAccrued();
+
+        if (fees > newAvailableAssets) return 0;
+
+        // Calculate the performance fee and remove from the available assets
+        return newAvailableAssets - fees;
+    }
+
+    /// @dev Calculate the available assets which is the assets in the ARM, external withdrawal queue,
+    /// less liquidity assets reserved for the ARM's withdrawal queue.
+    /// This does not exclude any accrued performance fees.
+    function _availableAssets() internal view returns (uint256) {
+        // Get the assets in the ARM and external withdrawal queue
+        uint256 assets = token0.balanceOf(address(this)) + token1.balanceOf(address(this)) + _externalWithdrawQueue();
+
+        // Load the queue metadata from storage into memory
+        uint256 queuedMem = withdrawsQueued;
```

Comment:
⚪️ We can make this method cleaner, clearer, and a tiny bit less gas by using the same style as `_liquidtyAvailable`

Just load from storage with:

```
uint256 outstandingWithdrawals = withdrawsQueued - withdrawsClaimed;
```

and then the rest of the function gets simpler using `outstandingWithdrawals` rather than math..

Reply:
that's a nice cleanup. Done